src/struct.rs

```diff
@@ -75,6 +75,12 @@ impl Struct {
         self
     }
 
+    /// Add an arbitrary macro.
+    pub fn r#macro(&mut self, r#macro: &str) -> &mut Self {
+        self.type_def.r#macro(r#macro);
+        self
+    }
+
     /// Push a named field to the struct.
```

Comment:
I'm using this commit generating a file to be used along with `modular-bitfield` crate. It has structs annotated in the attribute style, e.g. `#[bitfield]`. For this, i need to use `struct.r#macro("#[bitfield]")`. To remain consistent with other codegen annotation methods, would it be valuable to use auto-wrap? (e.g. `struct.r#macro("bitfield")`)

This just means changing line 80 to `self.type_def.r#macro(format!("#[{}]", r#macro);`

I can see one drawback being that it might make the library a touch more fragile overall.